src/Client.ts

```diff
@@ -1,56 +1,130 @@
-    private _presence:presenceClient.Presence;
-    private _ringout:ringoutClient.RingOut;
-
-    constructor(sdk) {
-
-        this._sdk = sdk;
-
-        this._account = new accountClient.Account(sdk);
-        this._versioning = new versioningClient.APIVersions(sdk);
-        this._callLog = new callLogClient.CallLog(sdk);
-        this._dictionary = new dictionaryClient.Dictionary(sdk);
-        this._extension = new extensionClient.Extension(sdk);
-        this._messages = new messagesClient.Messages(sdk);
-        this._notifications = new notificationsClient.NotificationsSubscriptionAPI(sdk);
-        this._presence = new presenceClient.Presence(sdk);
-        this._ringout = new ringoutClient.RingOut(sdk);
-
-    }
-
-    account() { return this._account; }
-    versioning() { return this._versioning; }
-    callLog() { return this._callLog; }
-    dictionary() { return this._dictionary; }
-    extension() { return this._extension; }
-    messages() { return this._messages; }
-    notifications() { return this._notifications; }
-    presence() { return this._presence; }
-    ringout() { return this._ringout; }
+import { parse as parseUrl } from "url";
+import Account from "./paths/Account";
+import ClientInfo from "./paths/ClientInfo";
+import NumberPool from "./paths/NumberPool";
+import NumberParser from "./paths/NumberParser";
+import Dictionary from "./paths/Dictionary";
+import Subscription from "./paths/Subscription";
+import * as Ringcentral from "ringcentral";
+
+export default class Client {
+
+    service: Ringcentral;
+
+    constructor(sv: Ringcentral) {
+        this.service = sv;
+    }
+
+    /**
+     * FIXME: Allow additional options: brand_id, display, prompte?
+     */
+    loginUrl(opts: {
+        redirectUri: string,
+        state?: string
+    }): string {
+        return this.service.platform().loginUrl(opts);
+    }
+
+    login(opts: {
+        // Login by password
+        username?: string;
+        password?: string;
+        extension?: string;
+
+        // Login by oauth
+        code?: string;
+        redirectUri?: string;
+
+        // Common options
+        accessTokenTtl?: number;
+        refreshTokenTtl?: number;
+        scope?: string[];
+    }): Promise<void> {
+        return this.service.platform().login(opts);
+    }
+
+    getAuthCode(callbackUrl: string): string {
+        let res = parseUrl(callbackUrl, true).query;
+        let code = res["code"];
+        if (code) {
+            return code;
+        } else if (res["error_description"]) {
+            let err = new Error(res["error_description"]);
+            err.name = res["error"];
+            err["errors"] = res["errors"];
+            throw err;
+        }
+    }
+
+    logout(): Promise<void> {
+        return this.service.platform().logout();
+    }
+
+    on(evtName: string, handler: Function) {
+        this.service.platform().on(evtName, handler);
+    }
```

Comment:
Can you explain more about why can't we provide such sugar methods? The instance of `RingCentralClient` will be passed around and users usually only keep reference of `RingCentralClient` instance not js-sdk, so it'll be of great convenience for users to write something like:

```javascript
client.logout();
client.on(name, handler);
```

Instead of writing longer code:
```javascript
client.service.platform().logout();
client.service.platform().on(name, handler);
```